test/test_browser.py

```diff
@@ -5825,6 +5825,25 @@ def test_webpack(self):
     shutil.copyfile('webpack/src/hello.wasm', 'webpack/dist/hello.wasm')
     self.run_browser('webpack/dist/index.html', '/report_result?exit:0')
 
+  def test_exceptions_stack_trace_and_message_wasm_eh(self):
+    # Currently writing to WebAssembly.Exception.stack property is not allowed
+    # in Firefox, but we need to ensure generating stack traces runs on FireFox
+    # without erroring out, even though stack trace simplification will not be
+    # done. See https://github.com/emscripten-core/emscripten/pull/21050
+    if not is_firefox():
+      self.skipTest('https://github.com/emscripten-core/emscripten/pull/21050')
+
+    create_file('main.cpp', r'''
+      #include <stdexcept>
+      int main() {
+        throw std::runtime_error("my message");
+        return 0;
+      }
+                ''')
+    self.emcc_args = ['-g', '-fwasm-exceptions']
+    self.set_setting('EXCEPTION_STACK_TRACES', 1)
+    self.btest_exit('main.c')
```

Comment:
Not a big deal but I wonder if this test could run the spidermonkey command line vm (and be part of test_core.py)?  

Looking at our circleci config it looks like we already run `test_demangle_stacks_symbol_map` under spidermonkey.

If you want to try doing this locally you can use `jsvu` to install spidermonkey and do something like this your `.emscripten`:

```
SPIDERMONKEY_ENGINE = [os.path.expanduser('~/.jsvu/bin/spidermonkey')]           
JS_ENGINES = [SPIDERMONKEY_ENGINE]
```

Reply:
OK this gets tricky... Yeah I installed spidermonkey and moved the test to `test_core.py`, with which I added a few functions and decorators for spidermonkey.

One funny thing is spidermonkey doesn't even seem to return anything with `Object.getOwnPropertyDescriptor(e, 'stack'); `, so it is not possible to check if `writable` is set. So I had to split it again into two `if`s:
```js
    if (e.stack) {
      var pd = Object.getOwnPropertyDescriptor(e, 'stack');
      if (pd && pd.writable) {
        var arr = e.stack.split('\n');
        arr.splice(1,1);
        e.stack = arr.join('\n');
      }
    }
```

But the funny thing is, my spidermonkey doesn't seem to print
```console
TypeError: setting getter-only property "stack"
```
as in #21044. It just ignores whatever I set it to, but doesn't error out:
```console
js> var tag = new WebAssembly.Tag({parameters: ["i32"]});
js> var ex = new WebAssembly.Exception(tag, [42], {traceStack: true});
js> ex.stack
"@typein:2:10\n"
js> ex.stack = "abc";
"abc"
js> ex.stack
"@typein:2:10\n"
```

My spidermonkey version is
```console
aheejin@aheejin:~/emscripten$ spidermonkey --version
JavaScript-C122.0
```

So, yeah, I updated the test but it is not really testing what I want to test, because the way I can think of to test this is to check if the error string contains `setting getter-only property "stack"` (because, due to exception, the test fails anyway), and now my spidermonkey doesn't emit that string. Not sure if that's due to version differences...

Reply:
> spidermonkey doesn't even seem to return anything with `Object.getOwnPropertyDescriptor(e, 'stack')`

That's because, like all WebIDL attributes, it's a prototype-placed accessor, not an own property. WebIDL `readonly` attributes are implemented as prototype-placed getters without corresponding setters. Chrome is in violation of the spec in using an own property instead.

As such the way to check is something like

```js
function propertyWritable(obj, prop) {
  let desc = Object.getOwnPropertyDescriptor(obj, prop);
  if (!desc) {
    let proto = Object.getPrototypeOf(obj);
    return proto == null || propertyWritable(proto, prop);
  }
  return "writable" in desc
    ? desc.writable
    : typeof desc.set === "function";
}

propertyWritable(new WebAssembly.Exception(new WebAssembly.Tag({parameters: ["i32"]}), [42], {traceStack: true}), 'stack'); // true in chrome, false in FF
```

Reply:
If setting `.stack` is not spec compliant then we should probably just not do it... even though its current possible on chrome.

Reply:
Having `stack` be writable isn't compliant, but if you do want to set the property, you could do `Object.defineProperty(e, 'stack', { value: whatever, configurable: true, writable: true })` and that would work even in browsers where `.stack` is non-writable. `defineProperty` works even for `readonly` properties.

(No comment on whether you should do this, just pointing out options.)

Reply:
Thanks for the info! But yeah, after all, I think it'd be better not to do this given that this is not spec compliant. Will close this in favor of #21065.